scripts/check_release_artifacts.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# should be run in CI or manually as:
+# ./check_release_artifacts.sh v2.3.1
+# DOCKER_USER and DOCKER_PASS need to be set for dockerhub checks when run manually
+set -e
+
+if [[ -n "$1" ]]; then
+    release=${1}
+else
+    echo specify release version e.g. ${0} v2.3.0 >&2; exit 1
+fi
+
+failed=false
+if [[ -n $release && $release =~ ^v([0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)*)$ ]]; then
+    version=${BASH_REMATCH[1]}
+else
+    version=$release
+fi
+
+declare -a targets=(ubuntu macos)
+repo_name=aeternity/aeternity
+
+check_release_asset() {
+    curl -fsS -o /dev/null --head "$1" 2>&1
+}
+
+check_github_release_assets() {
+    for target in "${targets[@]}"; do
+        GH_ASSET=https://github.com/$repo_name/releases/download/$release/aeternity-$version-$target-x86_64.tar.gz
+        if [[ $(check_release_asset $GH_ASSET) != "" ]]; then
+            echo $GH_ASSET check failed >&2; failed=true
+        fi
+        # TODO: compare MD5 checksum with local build
+    done
+}
+
+get_dockerhub_token() {
+    curl --silent --user "${DOCKER_USER}:${DOCKER_PASS}" \
+        "https://auth.docker.io/token?scope=repository:$repo_name:pull&service=registry.docker.io" \
+        |jq -r .token
+}
+
+get_docker_image_digest() {
+    local image=$1
+    local tag=$2
+    local token=$3
+   curl \
+    --silent \
+    --header "Accept: application/vnd.docker.distribution.manifest.v2+json" \
+    --header "Authorization: Bearer $token" \
+    "https://index.docker.io/v2/$image/manifests/$tag" \
+    | jq -r '.config.digest'
+}
+
+get_docker_image_config() {
+  local image=$1
+  local token=$2
+  local digest=$3
+
+  curl \
+    --silent \
+    --location \
+    --header "Authorization: Bearer $token" \
+    "https://registry-1.docker.io/v2/$image/blobs/$digest" \
+    | jq -r '.container_config'
+}
+
+check_dockerhub_assets() {
```

Comment:
but whyyy !?? ;(

```
$ curl https://index.docker.io/v1/repositories/aeternity/aeternity/tags/v2.0.0
[]
dincho at Dinchos-Mac-mini in ~/Projects/aeternity/infrastructure on health_checks [$]
$ curl https://index.docker.io/v1/repositories/aeternity/aeternity/tags/v2.0.5
Tag not found
```

https://stackoverflow.com/questions/32113330/check-if-imagetag-combination-already-exists-on-docker-hub/37799618

Reply:
This is not used, I'll remove it. It could be used if we want to compare remote image digest with local build image.  (did not dig into this much, just wanted to see what it prints)

Reply:
fixed in 3c5e3db

Reply:
I don't think you got the idea, this check can be done with single GET request, you don't need tokens and multiple REST API calls.

Reply:
Nevermind, digest check is a good addition.